llvm/lib/Transforms/Vectorize/LoopVectorizationLegality.cpp

```diff
@@ -395,24 +395,25 @@ static bool isUniformLoopNest(Loop *Lp, Loop *OuterLp) {
   return true;
 }
 
-static Type *convertPointerToIntegerType(const DataLayout &DL, Type *Ty) {
+static IntegerType *getInductionIntegerTy(const DataLayout &DL, Type *Ty) {
+  assert(Ty->isIntOrPtrTy() && "Expected integer or pointer type");
+
   if (Ty->isPointerTy())
-    return DL.getIntPtrType(Ty);
+    return DL.getIntPtrType(Ty->getContext(), Ty->getPointerAddressSpace());
```

Comment:
Whilst this looks like sensible, it doesn't seem like a NFC change to me unless there is something that guarantees we will only ever call this function with address space 0? I think I'd prefer this patch to remain strictly NFC and leave it as

```
  return DL.getIntPtrType(Ty);
```

then in a separate patch if you want you can change this and another instance in LoopVectorize.cpp where we call `DL.getIntPtrType` assuming the default address space? It would also be good to have tests that expose cases where different address spaces lead to different integer types.